awscli/customizations/overridesslcommonname.py

```diff
@@ -0,0 +1,122 @@
+        "af-south-1": "af-south-1.elasticmapreduce.amazonaws.com",
+        "ap-east-1": "ap-east-1.elasticmapreduce.amazonaws.com",
+        "ap-northeast-1": "ap-northeast-1.elasticmapreduce.amazonaws.com",
+        "ap-northeast-2": "ap-northeast-2.elasticmapreduce.amazonaws.com",
+        "ap-northeast-3": "ap-northeast-3.elasticmapreduce.amazonaws.com",
+        "ap-south-1": "ap-south-1.elasticmapreduce.amazonaws.com",
+        "ap-southeast-1": "ap-southeast-1.elasticmapreduce.amazonaws.com",
+        "ap-southeast-2": "ap-southeast-2.elasticmapreduce.amazonaws.com",
+        "ap-southeast-3": "ap-southeast-3.elasticmapreduce.amazonaws.com",
+        "ca-central-1": "ca-central-1.elasticmapreduce.amazonaws.com",
+        "eu-north-1": "eu-north-1.elasticmapreduce.amazonaws.com",
+        "eu-south-1": "eu-south-1.elasticmapreduce.amazonaws.com",
+        "eu-west-1": "eu-west-1.elasticmapreduce.amazonaws.com",
+        "eu-west-2": "eu-west-2.elasticmapreduce.amazonaws.com",
+        "eu-west-3": "eu-west-3.elasticmapreduce.amazonaws.com",
+        "me-south-1": "me-south-1.elasticmapreduce.amazonaws.com",
+        "sa-east-1": "sa-east-1.elasticmapreduce.amazonaws.com",
+        "us-east-2": "us-east-2.elasticmapreduce.amazonaws.com",
+        "us-west-1": "us-west-1.elasticmapreduce.amazonaws.com",
+        "us-west-2": "us-west-2.elasticmapreduce.amazonaws.com",
+    },
+    "rds": {
+        "us-east-1": "rds.amazonaws.com",
+    },
+    "docdb": {
+        "us-east-1": "rds.amazonaws.com",
+    },
+    "neptune": {
+        "us-east-1": "rds.amazonaws.com",
+    },
+    "health": {
+        "aws-global": "health-us-east-1.amazonaws.com",
+        "af-south-1": "health-us-east-1.amazonaws.com",
+        "ap-east-1": "health-us-east-1.amazonaws.com",
+        "ap-northeast-1": "health-us-east-1.amazonaws.com",
+        "ap-northeast-2": "health-us-east-1.amazonaws.com",
+        "ap-northeast-3": "health-us-east-1.amazonaws.com",
+        "ap-south-1": "health-us-east-1.amazonaws.com",
+        "ap-southeast-1": "health-us-east-1.amazonaws.com",
+        "ap-southeast-2": "health-us-east-1.amazonaws.com",
+        "ap-southeast-3": "health-us-east-1.amazonaws.com",
+        "ca-central-1": "health-us-east-1.amazonaws.com",
+        "eu-central-1": "health-us-east-1.amazonaws.com",
+        "eu-north-1": "health.us-east-1.amazonaws.com",
+        "eu-south-1": "health.us-east-1.amazonaws.com",
+        "eu-west-1": "health.us-east-1.amazonaws.com",
+        "eu-west-2": "health.us-east-1.amazonaws.com",
+        "eu-west-3": "health.us-east-1.amazonaws.com",
+        "me-south-1": "health.us-east-1.amazonaws.com",
+        "sa-east-1": "health.us-east-1.amazonaws.com",
+        "us-east-1": "health.us-east-1.amazonaws.com",
+        "us-east-2": "health.us-east-1.amazonaws.com",
+        "us-west-1": "health.us-east-1.amazonaws.com",
+        "us-west-2": "health.us-east-1.amazonaws.com",
```

Comment:
So when running the CLI manually with these hard coded endpoints, I'm getting the following error for non-us-east-1 regions when using `health`:
```
$ aws health describe-events --region us-west-2

An error occurred (InvalidSignatureException) when calling the DescribeEvents operation: Credential should be scoped to a valid region. 
```

It looks like the reason this is happening is that `health` is defined as a non-regionalized service and thus has a default credentialScope of `us-east-1` for the `aws` partition. We are going to have to adjust the logic here so that we also set the region to the default credential scope for `health` as well as the endpoint url. I would not worry about generalizing it too much. It should be just:

If we find we need to set the endpoint url for health:

1. Call the `session.get_partition_for_region()` for the provided region to determine the partition we are working in.
2. Map the partition to the value representing the region of the default credential scope for the partition (i.e. determine if the region we need to set should be `us-east-1`, `cn-northwest-1`)
3. Set the region either on the `parsed_globals` or via the `session.set_config_variable()` method, whatever does the job of making sure we use the correct credential scope for the region.

Reply:
alright i shot up a fix. i was getting the same error before making the fix, but a different one. not sure how to handle it since it was a subscription error but as far as i can tell there are no subscriptions in aws health.